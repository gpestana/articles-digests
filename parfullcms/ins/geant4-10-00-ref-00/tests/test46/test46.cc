//
// ********************************************************************
// * License and Disclaimer                                           *
// *                                                                  *
// * The  Geant4 software  is  copyright of the Copyright Holders  of *
// * the Geant4 Collaboration.  It is provided  under  the terms  and *
// * conditions of the Geant4 Software License,  included in the file *
// * LICENSE and available at  http://cern.ch/geant4/license .  These *
// * include a list of copyright holders.                             *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.  Please see the license in the file  LICENSE  and URL above *
// * for the full disclaimer and the limitation of liability.         *
// *                                                                  *
// * This  code  implementation is the result of  the  scientific and *
// * technical work of the GEANT4 collaboration.                      *
// * By using,  copying,  modifying or  distributing the software (or *
// * any work based  on the software)  you  agree  to acknowledge its *
// * use  in  resulting  scientific  publications,  and indicate your *
// * acceptance of all terms of the Geant4 Software license.          *
// ********************************************************************
//
// $Id: test46.cc 66512 2012-12-19 10:26:52Z gcosmo $
//
// -------------------------------------------------------------
//      GEANT4 test46
//
//  Simplified combined CMS calorimeter 
//
//  Authors: A.Ivanchenko and V.Ivanchenko
//           September 2008
//
//  Modified: 
//
// -------------------------------------------------------------
//
//
//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......

#include "G4RunManager.hh"
#include "G4UImanager.hh"
#include "G4UIterminal.hh"
#include "G4UItcsh.hh"
#include "Randomize.hh"

#include "DetectorConstruction.hh"
#include "G4PhysListFactory.hh"
#include "G4VModularPhysicsList.hh"
#include "PrimaryGeneratorAction.hh"

#include "RunAction.hh"
#include "EventAction.hh"
#include "StackingAction.hh"
#include "G4Timer.hh"

#include "QGSPCMS_FTFP_BERT_EML.hh"
#include "QGSPCMS_FTFP_BERT_EML95msc93.hh"

#include "G4VisExecutive.hh"
#include "G4UIExecutive.hh"

//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......

int main(int argc,char** argv) {

  G4Timer timer;
  timer.Start();

  //choose the Random engine
  CLHEP::HepRandom::setTheEngine(new CLHEP::RanecuEngine());

  //Construct the default run manager
  G4RunManager * runManager = new G4RunManager();

  //set mandatory initialization classes
  runManager->SetUserInitialization(new DetectorConstruction());

  // Physics List name defined via environment variable
  // or the default test46 name is used
  G4VModularPhysicsList* phys = 0;
  G4String physName = "";
  char* path = getenv("PHYSLIST");
  if(path) { physName = G4String(path); }
  if("QGSP_FTFP_BERT_EML" == physName) { phys = new QGSPCMS_FTFP_BERT_EML(); }
  else if("QGSP_FTFP_BERT_EML95msc93" == physName) { 
    phys = new QGSPCMS_FTFP_BERT_EML95msc93(); }
  else if("" == physName) { physName = "FTFP_BERT_EMV"; }

  if(!phys) {
    G4PhysListFactory factory;
    phys = factory.GetReferencePhysList(physName);
    if(!phys) { exit(1); }
  }

  runManager->SetUserInitialization(phys);

  runManager->SetUserAction(new PrimaryGeneratorAction());

  //set user action classes
  runManager->SetUserAction(new RunAction());
  runManager->SetUserAction(new EventAction());
  runManager->SetUserAction(new StackingAction());

  // get the pointer to the User Interface manager
  G4UImanager* UI = G4UImanager::GetUIpointer();

  if (argc==1)   // Define UI terminal for interactive mode
    {
#ifdef G4VIS_USE
      //visualization manager
      G4VisExecutive* visManager = new G4VisExecutive;
      visManager->Initialize();
#endif

#ifdef G4UI_USE
      G4UIExecutive * ui = new G4UIExecutive(argc,argv);
      ui->SessionStart();
      delete ui;
#endif

#ifdef G4VIS_USE
     delete visManager;
#endif
    }
  else           // Batch mode
    {
      G4String command = "/control/execute ";
      G4String fileName = argv[1];
      UI->ApplyCommand(command+fileName);
    }

  //job termination
  delete runManager;
  timer.Stop();
  G4cout << timer << std::endl;

  return 0;
}

//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
