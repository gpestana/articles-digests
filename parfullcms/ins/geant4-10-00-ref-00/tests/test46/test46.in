#================================================
#     Macro file for test46
#================================================
/control/verbose 2
/run/verbose 1
/tracking/verbose 0
#
/testCalorim/nbins 100
/testCalorim/maxE 50. GeV
/testCalorim/factEcal 1.03 
/testCalorim/factHcal 100.0
#
/testhadr/PrintModulo 10
/run/initialize
/run/setCut 1 mm
/run/setCutForRegion HcalRegion 1 mm
/gun/particle pi-
/gun/energy 15. GeV
/run/beamOn 100
#



