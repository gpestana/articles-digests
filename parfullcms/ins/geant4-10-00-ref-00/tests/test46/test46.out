
        ############################################
        !!! WARNING - FPE detection is activated !!!
        ############################################

*************************************************************
 Geant4 version Name: geant4-09-04-cand-02    (17-December-2010)
                      Copyright : Geant4 Collaboration
                      Reference : NIM A 506 (2003), 250-303
                            WWW : http://cern.ch/geant4
*************************************************************

### NIST DataBase for Materials is used

<<< Geant4 Physics List engine packaging library: PACK 5.5
<<< Geant4 Physics List simulation engine: FTFP_BERT_EMV 1.1

/run/verbose 1
/tracking/verbose 0
#
/testCalorim/nbins 100
/testCalorim/maxE 50. GeV
/testCalorim/factEcal 1.03
/testCalorim/factHcal 100.0
#
/testhadr/PrintModulo 10
/run/initialize
### New geometry is constructed HCAL width(cm)= 46
### Adding tracking cuts for neutron  TimeCut(ns)= 10000  KinEnergyCut(MeV)= 0
/run/setCut 1 mm
/run/setCutForRegion HcalRegion 1 mm
/gun/particle pi-
/gun/energy 15. GeV
/run/beamOn 100

phot:   for  gamma, applyCuts: 1     SubType= 12
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
       PhotoElectric :   Emin=        0 eV       Emax=         10 TeV

compt:   for  gamma, applyCuts: 1     SubType= 13
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
       Klein-Nishina :   Emin=        0 eV       Emax=         10 TeV

conv:   for  gamma, applyCuts: 1     SubType= 14
      Lambda tables from 1.022 MeV to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
        BetheHeitler :   Emin=        0 eV       Emax=         10 TeV

msc:   for e-    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc93 :   Emin=        0 eV       Emax=         10 TeV

eIoni:   for  e-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.8, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
        MollerBhabha :   Emin=        0 eV       Emax=         10 TeV

eBrem:   for  e-    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      LPM flag: 1 for E > 1 GeV
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               eBrem :   Emin=        0 eV       Emax=          1 GeV  AngularGenUrban
            eBremLPM :   Emin=        1 GeV      Emax=         10 TeV  AngularGenUrban

eIoni:   for  e+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.8, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
        MollerBhabha :   Emin=        0 eV       Emax=         10 TeV

eBrem:   for  e+    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      LPM flag: 1 for E > 1 GeV
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               eBrem :   Emin=        0 eV       Emax=          1 GeV  AngularGenUrban
            eBremLPM :   Emin=        1 GeV      Emax=         10 TeV  AngularGenUrban

annihil:   for  e+, applyCuts: 1     SubType= 5
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            eplus2gg :   Emin=        0 eV       Emax=         10 TeV

msc:   for proton    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :   Emin=        0 eV       Emax=         10 TeV

hIoni:   for  proton    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :   Emin=        0 eV       Emax=          2 MeV
          BetheBloch :   Emin=        2 MeV      Emax=         10 TeV

hBrems:   for  proton    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               hBrem :   Emin=        0 eV       Emax=         10 TeV

hPairProd:   for  proton    SubType= 4
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
           hPairProd :   Emin=        0 eV       Emax=         10 TeV

msc:   for GenericIon    SubType= 10
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 0
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :   Emin=        0 eV       Emax=         10 TeV

ionIoni:   for  GenericIon    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 0.1, dRoverRange= 0.1, integral: 1, fluct: 1, linLossLimit= 0.02
      Stopping Power data for 17 ion/material pairs 
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            BraggIon :   Emin=        0 eV       Emax=          2 MeV
          BetheBloch :   Emin=        2 MeV      Emax=         10 TeV

hIoni:   for  anti_proton    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :   Emin=        0 eV       Emax=          2 MeV
          BetheBloch :   Emin=        2 MeV      Emax=         10 TeV

msc:   for kaon+    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :   Emin=        0 eV       Emax=         10 TeV

hIoni:   for  kaon+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :   Emin=        0 eV       Emax=    1.05231 MeV
          BetheBloch :   Emin=  1.05231 MeV      Emax=         10 TeV

hBrems:   for  kaon+    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               hBrem :   Emin=        0 eV       Emax=         10 TeV

hPairProd:   for  kaon+    SubType= 4
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
           hPairProd :   Emin=        0 eV       Emax=         10 TeV

hIoni:   for  kaon-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :   Emin=        0 eV       Emax=    1.05231 MeV
          BetheBloch :   Emin=  1.05231 MeV      Emax=         10 TeV

hBrems:   for  kaon-    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               hBrem :   Emin=        0 eV       Emax=         10 TeV

hPairProd:   for  kaon-    SubType= 4
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
           hPairProd :   Emin=        0 eV       Emax=         10 TeV

muMsc:   for mu+    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.2, step limit type: 0, lateralDisplacement: 1, polarAngleLimit(deg)= 11.4592
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
        WentzelVIUni :   Emin=        0 eV       Emax=         10 TeV

muIoni:   for  mu+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :   Emin=        0 eV       Emax=        200 keV
          BetheBloch :   Emin=      200 keV      Emax=          1 GeV
        MuBetheBloch :   Emin=        1 GeV      Emax=         10 TeV

muBrems:   for  mu+    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
              MuBrem :   Emin=        0 eV       Emax=         10 TeV

muPairProd:   for  mu+    SubType= 4
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          muPairProd :   Emin=        0 eV       Emax=         10 TeV

CoulombScat:   for  mu+, applyCuts: 1     SubType= 1
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      11.4592 < Theta(degree) < 180, Eth(MeV)= 10; pLimit(GeV^1)= 0.139531
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
  eCoulombScattering :   Emin=        0 eV       Emax=         10 TeV

muIoni:   for  mu-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :   Emin=        0 eV       Emax=        200 keV
          BetheBloch :   Emin=      200 keV      Emax=          1 GeV
        MuBetheBloch :   Emin=        1 GeV      Emax=         10 TeV

muBrems:   for  mu-    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
              MuBrem :   Emin=        0 eV       Emax=         10 TeV

muPairProd:   for  mu-    SubType= 4
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          muPairProd :   Emin=        0 eV       Emax=         10 TeV

CoulombScat:   for  mu-, applyCuts: 1     SubType= 1
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      11.4592 < Theta(degree) < 180, Eth(MeV)= 10; pLimit(GeV^1)= 0.139531
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
  eCoulombScattering :   Emin=        0 eV       Emax=         10 TeV

hIoni:   for  pi+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :   Emin=        0 eV       Emax=    297.505 keV
          BetheBloch :   Emin=  297.505 keV      Emax=         10 TeV

hBrems:   for  pi+    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               hBrem :   Emin=        0 eV       Emax=         10 TeV

hPairProd:   for  pi+    SubType= 4
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
           hPairProd :   Emin=        0 eV       Emax=         10 TeV

msc:   for pi-    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :   Emin=        0 eV       Emax=         10 TeV

hIoni:   for  pi-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :   Emin=        0 eV       Emax=    297.505 keV
          BetheBloch :   Emin=  297.505 keV      Emax=         10 TeV

hBrems:   for  pi-    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               hBrem :   Emin=        0 eV       Emax=         10 TeV

hPairProd:   for  pi-    SubType= 4
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
           hPairProd :   Emin=        0 eV       Emax=         10 TeV
============================================================================================
             HADRONIC PROCESSES SUMMARY (verbose level 1)

                     Hadronic Processes for <anti_neutron>
          hadElastic  Models:              hElasticLHEP: Emin(GeV)=     0  Emax(GeV)= 100000

                     Hadronic Processes for <anti_proton>
          hadElastic  Models:              hElasticLHEP: Emin(GeV)=     0  Emax(GeV)= 100000

                     Hadronic Processes for <e->
      ElectroNuclear  Models:                     CHIPS: Emin(GeV)=     0  Emax(GeV)= 30000

                     Hadronic Processes for <gamma>
     PhotonInelastic  Models:                     CHIPS: Emin(GeV)=     0  Emax(GeV)= 3.5
                                        TheoFSGenerator: Emin(GeV)=     3  Emax(GeV)= 100000

                     Hadronic Processes for <kaon+>
          hadElastic  Models:              hElasticLHEP: Emin(GeV)=     0  Emax(GeV)= 100000
   KaonPlusInelastic  Models:                      FTFP: Emin(GeV)=     4  Emax(GeV)= 100000
                                        Bertini Cascade: Emin(GeV)=     0  Emax(GeV)= 5

                     Hadronic Processes for <kaon->
          hadElastic  Models:              hElasticLHEP: Emin(GeV)=     0  Emax(GeV)= 100000
  KaonMinusInelastic  Models:                      FTFP: Emin(GeV)=     4  Emax(GeV)= 100000
                                        Bertini Cascade: Emin(GeV)=     0  Emax(GeV)= 5

                     Hadronic Processes for <lambda>
          hadElastic  Models:              hElasticLHEP: Emin(GeV)=     0  Emax(GeV)= 100000

                     Hadronic Processes for <mu->
          muMinusCaptureAtRest

                     Hadronic Processes for <neutron>
          hadElastic  Models:             hElasticCHIPS: Emin(GeV)=     0  Emax(GeV)= 100000
    NeutronInelastic  Models:                      FTFP: Emin(GeV)=     4  Emax(GeV)= 100000
                                        Bertini Cascade: Emin(GeV)=     0  Emax(GeV)= 5
            nCapture  Models:                G4LCapture: Emin(GeV)=     0  Emax(GeV)= 20000
            nFission  Models:                G4LFission: Emin(GeV)=     0  Emax(GeV)= 20000

                     Hadronic Processes for <pi+>
          hadElastic  Models:              hElasticLHEP: Emin(GeV)=     0  Emax(GeV)= 1
                                        hElasticGlauber: Emin(GeV)=     1  Emax(GeV)= 100000
   PionPlusInelastic  Models:                      FTFP: Emin(GeV)=     4  Emax(GeV)= 100000
                                        Bertini Cascade: Emin(GeV)=     0  Emax(GeV)= 5

                     Hadronic Processes for <pi->
          hadElastic  Models:              hElasticLHEP: Emin(GeV)=     0  Emax(GeV)= 1
                                        hElasticGlauber: Emin(GeV)=     1  Emax(GeV)= 100000
  PionMinusInelastic  Models:                      FTFP: Emin(GeV)=     4  Emax(GeV)= 100000
                                        Bertini Cascade: Emin(GeV)=     0  Emax(GeV)= 5

                     Hadronic Processes for <proton>
          hadElastic  Models:             hElasticCHIPS: Emin(GeV)=     0  Emax(GeV)= 100000
     ProtonInelastic  Models:                      FTFP: Emin(GeV)=     4  Emax(GeV)= 100000
                                        Bertini Cascade: Emin(GeV)=     0  Emax(GeV)= 5
============================================================================================

========= Table of registered couples ==============================

Index : 0     used in the geometry : Yes     recalculation needed : No 
 Material : G4_AIR
 Range cuts        :  gamma  1 mm     e-  1 mm     e+  1 mm  proton 1 mm 
 Energy thresholds :  gamma  990 eV     e-  990 eV     e+  990 eV  proton 100 keV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 1     used in the geometry : Yes     recalculation needed : No 
 Material : G4_PbWO4
 Range cuts        :  gamma  1 mm     e-  1 mm     e+  1 mm  proton 1 mm 
 Energy thresholds :  gamma  89.3347 keV    e-  1.12535 MeV    e+  1.05931 MeV proton 100 keV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 2     used in the geometry : Yes     recalculation needed : No 
 Material : G4_Fe
 Range cuts        :  gamma  1 mm     e-  1 mm     e+  1 mm  proton 1 mm 
 Energy thresholds :  gamma  20.6438 keV    e-  1.29592 MeV    e+  1.21169 MeV proton 100 keV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 3     used in the geometry : Yes     recalculation needed : No 
 Material : G4_POLYSTYRENE
 Range cuts        :  gamma  1 mm     e-  1 mm     e+  1 mm  proton 1 mm 
 Energy thresholds :  gamma  2.4362 keV    e-  361.464 keV    e+  351.877 keV proton 100 keV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 4     used in the geometry : Yes     recalculation needed : No 
 Material : CuZn
 Range cuts        :  gamma  1 mm     e-  1 mm     e+  1 mm  proton 1 mm 
 Energy thresholds :  gamma  25.0857 keV    e-  1.386 MeV    e+  1.29592 MeV proton 100 keV
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

Index : 5     used in the geometry : Yes     recalculation needed : No 
 Material : G4_Fe
 Range cuts        :  gamma  1 mm     e-  1 mm     e+  1 mm  proton 1 mm 
 Energy thresholds :  gamma  20.6438 keV    e-  1.29592 MeV    e+  1.21169 MeV proton 100 keV
 Region(s) which use this couple : 
    HcalRegion

Index : 6     used in the geometry : Yes     recalculation needed : No 
 Material : CuZn
 Range cuts        :  gamma  1 mm     e-  1 mm     e+  1 mm  proton 1 mm 
 Energy thresholds :  gamma  25.0857 keV    e-  1.386 MeV    e+  1.29592 MeV proton 100 keV
 Region(s) which use this couple : 
    HcalRegion

====================================================================

### Run 0 start
### Histo books 13 histograms in <test46.root>
[Histo::book] File created: test46.root
Start Run processing.
EventAction: Event # 0 started
### G4EmSaturation::FindBirksCoefficient Birks coefficient for G4_PbWO4  0.008415 mm/MeV
### G4EmSaturation::FindBirksCoefficient Birks coefficient for G4_POLYSTYRENE  0.07943 mm/MeV
EventAction: Event # 10 started
EventAction: Event # 20 started
EventAction: Event # 30 started
EventAction: Event # 40 started
EventAction: Event # 50 started
EventAction: Event # 60 started
EventAction: Event # 70 started
EventAction: Event # 80 started
EventAction: Event # 90 started
Run terminated.
Run Summary
  Number of events processed : 100
  User=39.11s Real=39.51s Sys=0.03s
RunAction: End of run actions are started
HistoManager: End of run actions are started
========================================================
Beam particle                           pi-
Beam Energy(MeV)                        15000
Number of events                        100
Average number of MIPS (Edep < 0.8 GeV) 0.36
Average number of steps                 1.776e+04
Average number of neutron steps         6443
Average number of hadron steps          2282
Average number of gamma steps           1.132e+04
Average number of e+- steps             2961
==============  ECAL  ====================================
  1x1 (GeV) =      2.221 +-     0.2468    RMS= 2.468
  3x3 (GeV) =      3.447 +-      0.342    RMS= 3.42
  5x5 (GeV) =      3.841 +-     0.3714    RMS= 3.714
  E1/E9     =     0.7351 +-    0.02075    RMS= 0.2075
  E1/E25    =     0.6853 +-    0.02325    RMS= 0.2325
  E9/E25    =     0.9157 +-   0.007448    RMS= 0.07448
==============  HCAL  ====================================
Visible HCAL Edep(GeV)     =   0.06861
Visible HCAL e-Edep(GeV)   =   0.03503
Total HCAL Edep(GeV)       =   7.645 +- 0.1836
==========================================================
Erec(GeV)   =      11.16 +-        0.3    RMS= 3
Etot(GeV)   =      11.52 +-     0.1836    RMS= 1.836
==========================================================
Ecal/E0=   0.2537  RMS/E0(%)= 24.53
Hcal/E0=        0.505 +- 0.01213
Erec/E0=       0.7371 +- 0.01982
Etot/E0=       0.7609 +- 0.01213
==========================================================

[Histo::save] ROOT: files writing...
[Histo::save] ROOT: files closing...
#
G4 kernel has come to Quit state.
User=40.93s Real=41.37s Sys=0.05s
